operator/k8s_cep_gc.go

```diff
@@ -90,17 +91,41 @@ func enableCiliumEndpointSyncGC(once bool) {
 					// will delete all CEPs in the cluster regardless of the pod
 					// state therefore we won't even watch for the pod store.
 					if !once {
-						podObj, exists, err := watchers.PodStore.GetByKey(cepFullName)
-						if err != nil {
-							scopedLog.WithError(err).Warn("Unable to get pod from store")
-							continue
+						var podObj interface{}
+						var err error
+						exists := false
+						podChecked := false
+						for _, owner := range cep.ObjectMeta.OwnerReferences {
+							switch owner.Kind {
+							case "Pod":
+								podObj, exists, err = watchers.PodStore.GetByKey(cepFullName)
+								if err != nil {
+									scopedLog.WithError(err).Warn("Unable to get pod from store")
+								}
+								podChecked = true
+							case "CiliumNode":
+								podObj, exists, err = CiliumNodeStore.GetByKey(owner.Name)
```

Comment:
Given the previous suggested change regarding `CiliumNodeStore`:
```suggestion
								podObj, exists, err = ciliumNodeStore.GetByKey(owner.Name)
```

Reply:
Done.